src/include/txn_inline.h

```diff
@@ -224,6 +225,19 @@ __txn_next_op(WT_SESSION_IMPL *session, WT_TXN_OP **opp)
     op = &txn->mod[txn->mod_count++];
     WT_CLEAR(*op);
     op->btree = S2BT(session);
+
+    /*
+     * Store the ID of the latest transaction that is making an update. It can be used to determine
+     * if there is an active transaction on the btree. Retry if we race with another thread, but
+     * give up if that update is newer than this one.
+     */
+    btree_txn_id_prev = op->btree->max_upd_txn;
+    while (WT_TXNID_LT(btree_txn_id_prev, txn->id)) {
```

Comment:
I believe the atomic operation means that the `txn->id` in here is re-read each time through the loop. @luke-pearson - is that true and best practice in our code at the moment?

Reply:
Does this violate the assumption in WiredTiger that a write operation only takes one latch? @agorrod

Reply:
An atomic operation is different to a latch - so no this is OK.

Reply:
It should reload the value of `txn->id` for each loop iteration. The loop itself has changed since this comment was added though so it no longer applies. I think that is the best practice for these `cas` style loops assuming the `cas` itself cannot be built into the `while` condition.